.github/workflows/publish-helminput-plugin-component.yaml

```diff
@@ -0,0 +1,255 @@
+name: Publish Helm Input Plugin Component
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'bindings/go/helm/**'
+      - '.github/workflows/publish-helminput-plugin-component.yaml'
+    tags:
+      - 'bindings/go/helm/v*'
+  pull_request:
+    branches:
+      - main
+    paths:
+      - 'bindings/go/helm/**'
+      - '.github/workflows/publish-helminput-plugin-component.yaml'
+  workflow_dispatch:
+    inputs:
+      version:
+        description: "Component version, e.g., 0.0.1 or 0.0.1-alpha1. Must match existing bindings/go/helm/vX.Y.Z[-suffix] tag without the 'v'. If not set, 'main' is used."
+        required: false
+        type: string
+        default: "main"
+      # Add possible inputs for OCM repository and CLI image to be able to test with different repos/images
+      ocm_repository:
+        description: "OCM repository to publish to. Default is ghcr.io/open-component-model/plugins."
+        required: false
+        type: string
+        default: "ghcr.io/open-component-model/plugins"
+      ocm_image:
+        description: "OCM CLI image to use for publishing. Default is ghcr.io/open-component-model/cli:main"
+        required: false
+        type: string
+        default: "ghcr.io/open-component-model/cli:main"
+
+jobs:
+  build:
+    name: Build Plugin Component
+    runs-on: ubuntu-latest
+    outputs:
+      should_publish: ${{ steps.publish-check.outputs.should_publish }}
+      version: ${{ steps.calculate_version.outputs.version }}
+    permissions:
+      contents: read
+      id-token: write
+      attestations: write
+    steps:
+      - name: Determine checkout ref
+        id: checkout_ref
```

Comment:
i dont really understand why we need this. IMO the default ref on PRs should be the empty string and does not need explicit eval. On tag versions it should be the tag.